ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingPropertySource.java

```diff
@@ -276,30 +251,24 @@ public void createContents(Composite parent, boolean enabled) {
 			editor.create(parent, enabled);
 			editor.initialize();
 			fDescriptors.add(editor);
-			if (bordersNeeded)
+			if (bordersNeeded) {
 				fBlock.getToolkit().paintBordersFor(parent);
+			}
 		}
 	}
 
-	/**
-	 */
+	private static final String[] BOOLEAN_CHOICES = { "false", "true" }; //$NON-NLS-1$ //$NON-NLS-2$
+
 	public void save() {
 		String pid = fModel.getPluginBase().getId();
-		for (Enumeration<String> keys = fValues.keys(); keys.hasMoreElements();) {
-			String shortKey = keys.nextElement();
-			Object value = fValues.get(shortKey);
-			String svalue = value.toString();
-			if (value instanceof Integer)
-				svalue = fBooleanChoices[((Integer) value).intValue()];
-			IPath path = IPath.fromOSString(pid).append(shortKey);
-			fMasterOptions.setProperty(path.toString(), svalue);
-		}
+		fValues.forEach((key, value) -> {
+			String svalue = value instanceof Integer integer ? BOOLEAN_CHOICES[integer] : value.toString();
```

Comment:
I would extract 2 new private methods:
* `intToBoolean`
* `booleanToInt`

You could then make these 2 transformations consistent to each other and that would also improve readability and avoid obscure errors in the future (a typical one would be that someone flips this `if`-logic and suddenly `0 == true` in some other part of the code)

Reply:
As commented above I think this overall story is just simplified if Boolean objects are stored in the map instead of mapping boolean values to Integer/int values. Now clue why this choice was made before, but in the current form of the code I see no reason why not to use Boolean objects in stead.
Then all this is not necessary.

Reply:
See my previous comment about this probably being a **breaking change**